src/form-logic.rst

```diff
@@ -70,14 +69,84 @@ as well as any column that accepts an :ref:`expression <expressions>`.
   text, your_name, What is your name?
   note, hello_name, "Hello, ${your_name}."
 
+You can also refer to the current question or to the current question's parent group or repeat:
+
+.. csv-table::
+  :header: , Explanation, Example, Notes
+  
+  ., current question's value, . >= 18, Used in :ref:`constraints <constraints>`.
+  \.\., current question's parent group, position(..), Used with :func:`position` to get the iteration index.
+
+.. _xpath-paths:
+
+Advanced: XPath paths
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``${}`` notation in XLSForm is a convenient shortcut to refer to a specific field. When an XLSForm is converted, ``${}`` references are expanded to XPath paths which specifically describe where the field is located in the form.
+
+Some tools like ODK Build do not support ``${}`` notation so raw XPath notation must be used. Even in XLSForm, it can be advantageous to use raw XPath notation, especially in the context of repeats or to query datasets. The ``${}`` and XPath notations can be mixed freely.
+
+One way to think about XPath is that it sees a form or dataset like a series of folders and files on your computer. :ref:`Groups <groups>` and :ref:`repeats <repeats>` are like folders because they can contain other elements. Fields are like files and can't contain any other elements. Imagine a form with a group with name ``my_group`` which contains another group with name ``inner_group`` which contains a question with name ``q1``. The absolute path to ``q1`` is ``/data/my_group/inner_group/q1``.
+
+The ``/data/`` in the example above indicates that this is an absolute path for an element of the form itself (as opposed to a dataset, for example). This root is named ``data`` by default but can be modified by adding a :th:`name` column in the XLSForm **settings** sheet and specifying a value below it. This is rarely needed.
+
+XPath paths can also be relative. For example, let's say there's a ``relevance`` expression for ``q1`` in the example above and that this expression refers to a question with name ``eligible`` in the ``my_group`` group. We could refer to it using an absolute expression: ``/data/my_group/eligible``. We could also write a relative expression: ``../../eligible``. The ``..`` references the parent of the current node. Since our expression is on the node ``/data/my_group/inner_group/q1``, the first ``..`` references the group ``/data/my_group/inner_group``. We want to access a node in the ``my_group`` group so we need to go up one more level with another ``..``. The XPath path ``../../`` when used in an expression for the ``/data/my_group/inner_group/q1`` node references ``/data/my_group``. We can now go into ``my_group`` and reference the question with name ``eligible``: ``../../eligible``. Relative expressions are particularly useful in the context of repeats.
```

Comment:
I found this section very hard to parse. Here's my attempt...

> XPath paths can also be relative. For example, let's say there's a ``relevance`` expression for ``q1`` in the example above and that this expression refers to a ``group_name`` question in ``outer_group``. We could refer to ``group_name`` using an absolute expression: ``/data/outer_group/group_name``, but we could also write the relative expression ``../../group_name`` on ``q1``.
>
> The relative expression says that from your current position (`q1`), go up two levels (``../../``), and then reference ``group_name``. So in this case, the rightmost ``..`` references the parent of ``q1`` (``/data/outer_group/inner_group``) and leftmost ``..`` references the parent of that ``inner_group`` node (``/data/outer_group``). Once we have the ``outer_group``, we can reference the ``group_name`` question directly. The `/` is used to separate the different levels.

Note that I don't use ``eligible`` because it sounds like a keyword like ``relevance``.